internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -117,39 +122,16 @@ type CreateOpts struct {
 	Client               secret_service.ClientService
 }
 
-type SecretConfig struct {
-	Version         string
-	Type            integrations.IntegrationType
-	IntegrationName string `yaml:"integration_name"`
-	Details         map[string]any
-}
-
-type MongoDBRole struct {
-	RoleName       string `mapstructure:"role_name"`
-	DatabaseName   string `mapstructure:"database_name"`
-	CollectionName string `mapstructure:"collection_name"`
-}
-
-type MongoDBScope struct {
-	Name string `mapstructure:"type"`
-	Type string `mapstructure:"name"`
-}
-
-type AwsAssumeRole struct {
-	RoleArn string `mapstructure:"role_arn"`
+type detailsInternal struct {
+	Details map[string]any
 }
 
-type GcpServiceAccount struct {
-	ServiceAccountEmail string `mapstructure:"service_account_email"`
+type SecretConfig struct {
+	Type            integrations.IntegrationType `hcl:"type"`
+	IntegrationName string                       `hcl:"integration_name"`
+	Details         cty.Value                    `hcl:"details"`
```

Comment:
If the `Details` was a `map[string]interface{}` instead of `cty.Value`, would the HCL parser be able to decode it correctly (eliminating the need for custom recursive decoding logic in `ctyValueToMap`)?

Reply:
HCL[ unfortunately does not support `map[string]interface{}`](https://github.com/hashicorp/hcl/issues/291) since it requires defined types. `cty.Value` (or something like `hcl-Attributes`) is the closest thing to the flexibility we need.